### DISCUSSION TITLE
Add Option to Block DMs

### DISCUSSION THREAD

**Discussion by vanhalf** [jakobdylanc/llmcord.py#80] [Ideas]
Hi,

It would be great to have an option to block DMs. On my server, a lot of people have started sending unsolicited messages directly to the bot via DMs instead of writing in the server channels.

Here’s an example of how this could be implemented:

In llmcord.py:
```
allowed_dms = cfg["allowed_dms"]

if not allowed_dms and new_msg.guild is None:
    logging.info(f"Message in DM from {new_msg.author.id}: {new_msg.content}")
    return
```
In config.yaml:
```
allowed_dms: false
```

This approach checks the allowed_dms configuration and only processes DMs if it's enabled. It can help reduce unwanted bot interactions in private messages.
Thank you for considering this suggestion!

**Comment by jakobdylanc** [comment]
I've hesitated adding this since you can already disable DMs by using `allowed_channel_ids` or `allowed_role_ids`. But I agree, a simple true/false setting for DMs would still be nice.

I'll probably add this soon, just have to figure out some small details. Thanks for the suggestion!

**Comment by jakobdylanc** [comment]
Done - https://github.com/jakobdylanc/llmcord/commit/714e195f277aa6483b1dd4756a5756648304ffc1

**Reply by vanhalf** [reply]
great :)
